Assets/Scripts/Managers/SimManager.cs

```diff
@@ -394,7 +394,7 @@ public Interceptor CreateInterceptor(Configs.AgentConfig config, Simulation.Stat
     // Create a sensor for the interceptor.
     switch (config.DynamicConfig.SensorConfig.Type) {
       case Simulation.SensorType.Ideal: {
-        interceptorObject.AddComponent<IdealSensor>();
+        interceptorObject.AddComponent<IdealSensorLegacy>();
         break;
       }
       default: {
```

Comment:
Do we want to consider a MonoBehaviour wrapper? e.g.
```cs
public class IdealSensorComponent : MonoBehaviour {
    private IdealSensor _sensor;

    private void Awake() {
      var agent = GetComponent<IAgent>();
      _sensor = new IdealSensor(agent);
    }

    public SensorOutput Sense(IHierarchical hierarchical) => _sensor.Sense(hierarchical);
    public SensorOutput Sense(in Vector3 waypoint) => _sensor.Sense(waypoint);
}
```

Or shall we store sensors as fields?
```cs
// In Interceptor or Agent class
private ISensor _sensor;

void Initialize() {
_sensor = new IdealSensor(this);
}
```

Reply:
I'm leaning towards sensors as fields; I don't see a benefit to a wrapper, and it makes unit-testing them harder.